src/databricks/labs/blueprint/paths.py

```diff
@@ -448,17 +441,296 @@ def match(self, path_pattern, *, case_sensitive=None):
                 return False
         return True
 
-    def as_fuse(self):
+    def home(self):  # pylint: disable=arguments-differ
+        """Return the user's home directory. Adapted from pathlib.Path"""
+        return type(self)(self._ws, "~").expanduser()
+
+    @abstractmethod
+    def _rename(self: P, target: str | bytes | os.PathLike, overwrite: bool) -> P: ...
+
+    def rename(self: P, target: str | bytes | os.PathLike) -> P:
+        """Rename this path as the target, unless the target already exists."""
+        return self._rename(target, overwrite=False)
+
+    def replace(self: P, target: str | bytes | os.PathLike) -> P:
+        """Rename this path, overwriting the target if it exists and can be overwritten."""
+        return self._rename(target, overwrite=True)
+
+    def _return_false(self) -> bool:
+        return False
+
+    is_symlink = _return_false
+    is_block_device = _return_false
+    is_char_device = _return_false
+    is_fifo = _return_false
+    is_socket = _return_false
+    is_mount = _return_false
+    is_junction = _return_false
+
+    def resolve(self: P, strict: bool = False) -> P:
+        """Return the absolute path of the file or directory in Databricks Workspace."""
+        absolute = self.absolute()
+        if strict and not absolute.exists():
+            msg = f"Path does not exist: {self}"
+            raise FileNotFoundError(msg)
+        return absolute
+
+    def absolute(self: P) -> P:
+        if self.is_absolute():
+            return self
+        return self.with_segments(self.cwd(), self)
+
+    def _prepare_pattern(self, pattern: str | bytes | os.PathLike) -> Sequence[str]:
+        if not pattern:
+            raise ValueError("Glob pattern must not be empty.")
+        parsed_pattern = self.with_segments(pattern)
+        if parsed_pattern.anchor:
+            msg = f"Non-relative patterns are unsupported: {pattern!s}"
+            raise NotImplementedError(msg)
+        pattern_parts = parsed_pattern._path_parts  # pylint: disable=protected-access
+        if ".." in pattern_parts:
+            msg = f"Parent traversal is not supported: {pattern!s}"
+            raise ValueError(msg)
+        if os.fspath(pattern)[-1] == self.parser.sep:
+            pattern_parts = (*pattern_parts, "")
+        return pattern_parts
+
+    def glob(
+        self: P, pattern: str | bytes | os.PathLike, *, case_sensitive: bool | None = None
```

Comment:
```suggestion
        self: P, pattern: str | bytes | os.PathLike, *, case_sensitive: bool | None = None,
```
nit, so that we have line per argument